src/main/java/cs2103/v15_1j/jimjim/ui/TaskEventRowFactory.java

```diff
@@ -431,29 +457,29 @@ private boolean checkOverdue(DeadlineTask t){
 
 		return overdue;
 	}
-	
+
 	private boolean checkNotBefore(LocalDate date, TaskEvent taskEvent){
```

Comment:
very ugly to keep checking null and type like this. For null, just make sure to check before calling this function. For type, i think should have 2 overloaded versions